src/StreamJsonRpc.Shared/JsonRpc.cs

```diff
@@ -67,6 +67,11 @@ internal OutstandingCallData(object taskCompletionSource, Action<JsonRpcMessage>
         /// A delegate for the <see cref="HandleInvocationTaskResult(JToken, Task)"/> method.
         /// </summary>
         private readonly Func<Task, object, JsonRpcMessage> handleInvocationTaskResultDelegate;
+        
+        /// <summary>
+        /// A map of the target object's method names to MethodNameAttribute value.
+        /// </summary>
+        private readonly Dictionary<string, List<string>> methodToAttributeMap;
```

Comment:
It's not clear from the name what attribute it is about. Should it be named _methodNames_ instead?